bin.src/ingestLinearizeDecam.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+from __future__ import absolute_import, division, print_function
+"""Convert DECam official linearity FITS table into LSST linearity pickles
+
+To use, see UsageStr
+"""
+UsageStr = """Usage: ingestLinearizeDecam.py fromPath repoPath
+    where fromPath is a path to the DECam linearity FITS file
+    and repoPath is the path of the calib data repository
+"""
+import sys
+import os.path
+from itertools import izip
+
+import numpy as np
+import astropy.io.fits as fits
+
+from lsst.daf.persistence import Butler
+from lsst.ip.isr import LinearizeLookupTable
+
+def ingestLinearizeDecam(inPath, repoPath):
+    """Convert the specified DECam linearity FITS table to standard LSST format
+
+    Details:
+    - Input format is one table per CCD, HDU is amplifier number,
+        the table has 3 columns: ADU, ADU_LINEAR_A, ADU_LINEAR_B.
+        The values of ADU contiguous (0, 1, 2...) but check and error out if not.
+        The values of the latter two are replacements (0+delta0, 1+delta1, 2+delta2...)
+        and this is converted to offsets for the LSST linearization tables (delta0, delta1, delta2...)
+    - Output is to %(path)s/linearity/_%02(ccdnum)d.pickle
+    - The row indices for the linearization lookup table are (row index=amp name): 0=A, 1=B
+
+    @param[in] inPath  path to input table (one HDU per amplifier)
+    @param[in] butler  butler to output repo
+    """
+    print("Ingesting linearity tables from %r to %r" % (inPath, repoPath))
+    butler = Butler(root=repoPath)
+    camera = butler.get("camera")
+    fromHDUs = fits.open(inPath)[1:] # HDU 0 has no data
+    assert len(fromHDUs) == len(camera)
+    for ccdind, (detector, hdu) in enumerate(izip(camera, fromHDUs)):
+        ccdnum = ccdind + 1
+        print("ccdnum=%s" % (ccdnum,))
+        fromData = hdu.data
+        assert len(fromData.dtype) == 3
+        lsstTable = np.zeros((2, len(fromData)), dtype=np.float32)
+        uncorr = fromData["ADU"]
+        if not np.allclose(uncorr, np.arange(len(fromData))):
+            raise RuntimeError("ADU data not a range of integers starting at 0")
+        for i, ampName in enumerate("AB"):
+            # convert DECam replacement table to LSST offset table
+            print("DECam table for %s=%s..." % (ampName, fromData["ADU_LINEAR_" + ampName][0:5],))
+            lsstTable[i,:] = fromData["ADU_LINEAR_" + ampName] - uncorr
+            print("LSST  table for %s=%s..." % (ampName, lsstTable[i,0:5],))
+        linearizer = LinearizeLookupTable(table=lsstTable, detector=detector)
+        dataId = dict(path=repoPath, ccdnum=ccdnum)
```

Comment:
Good point. I had missed that change when you fixed the linearizer entry in the .paf file